src/components/form-component.js

```diff
@@ -197,12 +196,64 @@ function createFormClass(s = defaultStrategy) {
               validityChanged = true;
             }
 
+            // Changed the below for a test that errors and validations
+            // get merged correctly, but it appears this wasn't actually
+            // supported for the same field? Also could have the side
+            // effect that errors wouldn't get cleared?
+            // fieldsErrors[field] = merge(fieldsErrors[field] || {}, fieldErrors);
+
             fieldsErrors[field] = fieldErrors;
           }
         }
       };
 
-      mapValues(errorValidators, validateField);
+      // Run errors first, validations should take precendence.
+      // When run below will replace the contents of the fieldErrors[].
+      mapValues(errors, validateField);
+
+      if (typeof validators === 'function') {
+        const field = '';
+
+        const nextValue = field
+          ? s.get(nextProps.modelValue, field)
+          : nextProps.modelValue;
+
+        const currentValue = field
+          ? s.get(modelValue, field)
+          : modelValue;
+
+        // If the validators didn't change, the validity didn't change.
+        if ((!initial && !validatorsChanged) && (nextValue === currentValue)) {
+          // TODO this will only set the errors on form when using the function.
+          // How handle? Safe to assume will be no dispatch?
+          // fieldsErrors[field] = getField(formValue, field).errors;
```

Comment:
Yes, this is a safe assumption - do nothing if the values are the same after initial load.